src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/PersistenceService.java

```diff
@@ -158,6 +161,27 @@ public void deleteRoleAssignmentByActorId(UUID actorId) {
 
     }
 
+    public List<RoleAssignment> getAssignmentsByActorAndCaseId(String actorId, String caseId, String roleType) {
+        Set<RoleAssignmentEntity> roleAssignmentEntities = null;
+
+        if (StringUtils.isNotEmpty(actorId) && StringUtils.isNotEmpty(caseId)) {
+            roleAssignmentEntities = roleAssignmentRepository.findByActorIdAndCaseId(actorId, caseId, roleType);
+        } else if (StringUtils.isNotEmpty(actorId)) {
+            roleAssignmentEntities =
+                roleAssignmentRepository.findByActorIdAndRoleType(UUID.fromString(actorId), roleType);
+        } else if (StringUtils.isNotEmpty(caseId)) {
+            roleAssignmentEntities = roleAssignmentRepository.getAssignmentByCaseId(caseId, roleType);
+        }
+
+        if (roleAssignmentEntities == null || roleAssignmentEntities.isEmpty()) {
+            throw new ResourceNotFoundException(V1.Error.ASSIGNMENT_RECORDS_NOT_FOUND);
+        }
```

Comment:
Just check, whether it is good practice to return 404 from search query. It can return empty response if nothing is found.

Reply:
For now, returning 404. Doing research on this one.